src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_io_net_stm32f4_nucleo_wifi.c

```diff
@@ -0,0 +1,198 @@
+{
+    ( void )xi_socket;
+    ( void )host;
+    ( void )port;
+
+    char *protocol = "s";//t -> tcp , s-> secure tcp, c-> secure tcp with certs
+
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    status = wifi_socket_client_open((uint8_t *)"broker.dev.xively.io", port, (uint8_t *)protocol, (uint8_t *)xi_socket);
+
+    if ( status == WiFi_MODULE_SUCCESS ) return XI_BSP_IO_NET_STATE_OK;
+    else return XI_BSP_IO_NET_STATE_ERROR;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_connection_check( xi_bsp_socket_t xi_socket,
+                                                      const char* host,
+                                                      uint16_t port )
+{
+    ( void )xi_socket;
+    ( void )host;
+    ( void )port;
+
+    if ( xi_wifi_state == xi_wifi_state_connected ) return XI_BSP_IO_NET_STATE_OK;
+    else return XI_BSP_IO_NET_STATE_ERROR;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_write( xi_bsp_socket_t xi_socket,
+                                           int* out_written_count,
+                                           const uint8_t* buf,
+                                           size_t count )
+{
+    ( void )xi_socket;
+    ( void )buf;
+    *out_written_count = count;
+
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    status = wifi_socket_client_write(xi_socket, count, (char*) buf);
+
+    if ( status == WiFi_MODULE_SUCCESS ) return XI_BSP_IO_NET_STATE_OK;
+    else return XI_BSP_IO_NET_STATE_ERROR;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_read( xi_bsp_socket_t xi_socket,
+                                          int* out_read_count,
+                                          uint8_t* buf,
+                                          size_t count )
+{
+    ( void )xi_socket;
+    ( void )buf;
+    ( void )count;
+
+    *out_read_count = 0;
+
+    if ( xi_wifi_state == xi_wifi_state_connected )
+    {
+      if ( xi_wifi_buffer.size > 0 )
+      {
+          memcpy( buf , xi_wifi_buffer.bytes , xi_wifi_buffer.size );
+          *out_read_count = xi_wifi_buffer.size;
+          xi_wifi_buffer.size = 0;
+      }
+      else return XI_BSP_IO_NET_STATE_BUSY;
+    }
+    else if ( xi_wifi_state == xi_wifi_state_error )
+    {
+      return XI_BSP_IO_NET_STATE_ERROR;    
+    }
+    else if ( xi_wifi_state == xi_wifi_state_connected )
+    {
+      return XI_BSP_IO_NET_STATE_CONNECTION_RESET;    
+    }
+
+    return XI_BSP_IO_NET_STATE_OK;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_close_socket( xi_bsp_socket_t* xi_socket )
+{
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    status = wifi_socket_client_close( *xi_socket );
+
+    if ( status == WiFi_MODULE_SUCCESS ) return XI_BSP_IO_NET_STATE_OK;
+    else return XI_BSP_IO_NET_STATE_ERROR;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_select( xi_bsp_socket_events_t* socket_events_array,
+                                            size_t socket_events_array_size,
+                                            long timeout_sec )
+{
+    (void)timeout_sec;
```

Comment:
Note:

The select function has a timeout value - here it is not used, it means that the xively-client-c will actively wait instead of passive waiting, this is not good in terms of CPU usage and power usage etc. 

Question:
Is there no select or select like functionality for this WiFi extension  ?